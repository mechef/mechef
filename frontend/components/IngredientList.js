// @flow
import * as React from 'react';
import Rx from 'rxjs/Rx';

import Button from './Button';

type Props = {
  memos: Array<{
    _id: string,
    sum: number,
    name: string,
    ingredients: Array<{
      name: string,
      amount: number,
    }>,
  }>,
  onEditMemo: (memoId: string) => Rx.Observable,
}
const IngredientList = ({ memos, onEditMemo }: Props): React.Element<'div'> => (
  <div>
    <div className="header">
      <span className="title">Ingredients List</span>
      <div role="link" tabIndex="-1" className="addButton" onClick={() => onEditMemo('')}>
        <i className="fa fa-plus plus-icon" aria-hidden="true" />
      </div>
    </div>
    {
      memos.map(memo => (
        <div key={memo._id} className="ingredientList">
          <div className="ingredient-item">
            <div className="ingredient-content">
              <p className="ingredient-title">{memo.name}</p>
              <p className="ingredient-detail">
                <span className="ingredient-subtext">
                  Ingredient: {memo.ingredients && memo.ingredients.length}
                </span>
                <span className="ingredient-subtext">
                  Total: {`$${memo.sum}`}
                </span>
              </p>
            </div>
            <div className="buttonWrapper">
              <Button
                size="small"
                buttonStyle="primary"
                onClick={() => onEditMemo(memo._id)}
              >
                UPDATE
              </Button>
            </div>
          </div>
        </div>
      ))
    }
    <style jsx>
      {`
        .ingredientList {
          width: 800px;
        }
        .header {
          display: flex;
          align-items: center;
          padding-bottom: 22px;
        }
        .title {
          font-size: 18px;
          line-height: 1.11;
          letter-spacing: 0.5px;
          color: #4a4a4a;
        }
        .addButton {
          display: flex;
          width: 36px;
          height: 36px;
          margin-left: 20px;
          border-radius: 4px;
          background-color: #ffffff;
          cursor: pointer;
        }
        .plus-icon {
          margin: auto;
          color: #009245;
        }
        .addButton:hover {
          background-color: #3f9f40;
        }
        .addButton:hover .plus-icon {
          color: #ffffff;
        }
        .ingredient-item {
          display: flex;
          justify-content: space-between;
          margin-bottom: 12px;
          padding: 30px 20px 30px 15px;
          width: 100%;
          border-radius: 4px;
          background-color: #ffffff;
        }
        .ingredient-content {
          display: flex;
          flex-direction: column;
        }
        .ingredient-title {
          font-size: 16px;
          font-weight: 500;
          line-height: 1;
          letter-spacing: 0.7px;
          text-align: left;
          color: #4a4a4a;
        }
        .ingredient-detail {
          padding-top: 16px;
        }
        .ingredient-subtext {
          margin-right: 40px;
          font-size: 14px;
          font-weight: 500;
          line-height: 1.14;
          letter-spacing: 0.6px;
          text-align: left;
          color: #9b9b9b;
        }
        .buttonWrapper {
          align-self: center;
        }
      `}
    </style>
  </div>
);

export default IngredientList;
