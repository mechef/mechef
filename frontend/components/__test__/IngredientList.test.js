import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';

import IngredientList from '../IngredientList';

describe('Pages With Snapshot Testing', () => {
  it('snapshot IngredientList Component', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <IngredientList
        memos={[
          {
            _id: '59d054c8a70e5b90e75111b8',
            sum: '100',
            name: 'bible memo',
            ingredients: [
              {
                name: 'ingredient2',
                amount: '1234',
              },
            ],
          },
        ]}
        onEditMemo={() => {}}
        onDeleteMemo={() => {}}
        t={() => {}}
      />,
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
