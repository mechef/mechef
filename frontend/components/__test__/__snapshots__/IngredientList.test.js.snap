// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot IngredientList Component 1`] = `
<div
  className="jsx-1552839758"
>
  <div
    className="jsx-1552839758 header"
  >
    <span
      className="jsx-1552839758 title"
    >
      Ingredients List
    </span>
    <div
      className="jsx-1552839758 addButton"
      onClick={[Function]}
      role="link"
      tabIndex="-1"
    >
      <i
        aria-hidden="true"
        className="jsx-1552839758 fa fa-plus plus-icon"
      />
    </div>
  </div>
  <div
    className="jsx-1552839758 ingredientList"
  >
    <div
      className="jsx-1552839758 ingredient-item"
    >
      <div
        className="jsx-1552839758 ingredient-content"
      >
        <p
          className="jsx-1552839758 ingredient-title"
        >
          bible memo
        </p>
        <p
          className="jsx-1552839758 ingredient-detail"
        >
          <span
            className="jsx-1552839758 ingredient-subtext"
          >
            Ingredient: 
            1
          </span>
          <span
            className="jsx-1552839758 ingredient-subtext"
          >
            Total: 
            $100
          </span>
        </p>
      </div>
      <div
        className="jsx-1552839758 buttonWrapper"
      >
        <button
          className="jsx-675350272 
      button
      primary
      small
    "
          onClick={[Function]}
          type="button"
        >
          UPDATE
        </button>
      </div>
    </div>
  </div>
</div>
`;
